saleor/graphql/discount/types/promotions.py

```diff
@@ -99,3 +115,65 @@ def resolve_promotion(root: models.PromotionRule, info: ResolveInfo):
     @staticmethod
     def resolve_channels(root: models.PromotionRule, info: ResolveInfo):
         return ChannelsByPromotionRuleIdLoader(info.context).load(root.id)
+
+
+class PromotionEvent(ModelObjectType[models.PromotionEvent]):
```

Comment:
Wouldn't be better to have small types dedicated to specific actions instead of generic one?
The current type doesn't contain a lot of fields but probably it will going to grow in the future. We should consider using `UNION` here - like `PromotionCreatedEvent`, `PromotionRuleCrated` etc. 
With union here, we give frontend apps more context on which fields are related to specific events. 
With the current approach, frontend apps need to know how to combine the fields. For example by calling
```graphql
{
    promotions{
        events{
            ruleId
        }
    }
}
```
Frontend needs to be aware that the `ruleId` field will have a value only for specific type of events (related to the rules).  Also frontend needs to be aware that `ruleId` doesn't provide here any value when type of the event is `PROMOTION_CREATED`